providers/hubspot/read.go

```diff
@@ -26,28 +32,11 @@ func (c *Connector) Read(ctx context.Context, config common.ReadParams) (*common
 		err error
 	)
 
-	// If filtering is required, then we have to use the search endpoint.
-	// The Search endpoint has a 10K record limit. In case this limit is reached,
-	// the sorting allows the caller to continue in another call by offsetting
-	// until the ID of the last record that was successfully fetched.
-	if requiresFiltering(config) {
-		searchParams := SearchParams{
-			ObjectName: config.ObjectName,
-			FilterGroups: []FilterGroup{
-				{
-					Filters: []Filter{
-						BuildLastModifiedFilterGroup(&config),
-						// Add more filters to AND them together
-					},
-					// Add more filter groups to OR them together
-				},
-			},
-			SortBy: []SortBy{
-				BuildSort(ObjectFieldHsObjectId, SortDirectionAsc),
-			},
-			NextPage: config.NextPage,
-			Fields:   config.Fields,
-		}
+	// If filtering is required, then we have to use the search endpoint. This is also the case when the query is
+	// too long. The Search endpoint has a 10K record limit. In case this limit is reached, the sorting allows the
+	// caller to continue in another call by offsetting until the ID of the last record that was successfully fetched.
```

Comment:
So it sounds like only the search endpoint supports includeAllProperties, not the List endpoint? Can you add that to the comments if that's the case?

Also I'm forgetting why we even call List for backfills. Your statement "In case this limit is reached, the sorting allows the caller to continue in another call by offsetting until the ID of the last record that was successfully fetched." seems to indicate that we would be fine with calling the Search endpoint always, even for backfills?